examples/preprocessing/plot_scaling_importance.py

```diff
@@ -1,106 +1,149 @@
-=========================================================
-
-Feature scaling through standardization (or Z-score normalization)
-can be an important preprocessing step for many machine learning
-algorithms. Standardization involves rescaling the features such
-that they have the properties of a standard normal distribution
-with a mean of zero and a standard deviation of one.
-
-While many algorithms (such as SVM, K-nearest neighbors, and logistic
-regression) require features to be normalized, intuitively we can
-think of Principle Component Analysis (PCA) as being a prime example
-of when normalization is important. In PCA we are interested in the
-components that maximize the variance. If one component (e.g. human
-height) varies less than another (e.g. weight) because of their
-respective scales (meters vs. kilos), PCA might determine that the
-direction of maximal variance more closely corresponds with the
-'weight' axis, if those features are not scaled. As a change in
-height of one meter can be considered much more important than the
-change in weight of one kilogram, this is clearly incorrect.
-
-To illustrate this, :class:`PCA <sklearn.decomposition.PCA>`
-is performed comparing the use of data with
-:class:`StandardScaler <sklearn.preprocessing.StandardScaler>` applied,
-to unscaled data. The results are visualized and a clear difference noted.
-The 1st principal component in the unscaled set can be seen. It can be seen
-that feature #13 dominates the direction, being a whole two orders of
-magnitude above the other features. This is contrasted when observing
-the principal component for the scaled version of the data. In the scaled
-version, the orders of magnitude are roughly the same across all the features.
-
-The dataset used is the Wine Dataset available at UCI. This dataset
-has continuous features that are heterogeneous in scale due to differing
-properties that they measure (i.e. alcohol content and malic acid).
-
-The transformed data is then used to train a naive Bayes classifier, and a
-clear difference in prediction accuracies is observed wherein the dataset
-which is scaled before PCA vastly outperforms the unscaled version.
+=============================
 
-"""
-import matplotlib.pyplot as plt
+Feature scaling through standardization (also called Z-score normalization) can
+be an important preprocessing step for many machine learning algorithms. It
+involves rescaling each feature such that it has a standard deviation equals to
+1 and a mean equals to 0.
 
-from sklearn.model_selection import train_test_split
-from sklearn.preprocessing import StandardScaler
-from sklearn.decomposition import PCA
-from sklearn.naive_bayes import GaussianNB
-from sklearn.metrics import accuracy_score
-from sklearn.datasets import load_wine
-from sklearn.pipeline import make_pipeline
+Many algorithms require features to be normalized, either to ease the
+convergence (such as the logistic regression) or because fitting scaled data
```

Comment:
Penalized logreg is a very different model with and without scaling. It is not (only) a matter of easier optimization (convergence).